hack/lib/golang.sh

```diff
@@ -158,11 +163,20 @@ readonly KUBE_TEST_PORTABLE=(
 # need to target server platforms.
 # These binaries will be distributed in the kubernetes-test tarball.
 # If you update this list, please also update build/release-tars/BUILD.
-readonly KUBE_TEST_SERVER_TARGETS=(
-  cmd/kubemark
-  vendor/github.com/onsi/ginkgo/ginkgo
-  test/e2e_node/e2e_node.test
-)
+kube::golang::server_test_targets() {
+  local targets=(
+    cmd/kubemark
+    vendor/github.com/onsi/ginkgo/ginkgo
+  )
+
+  if [[ "${OSTYPE:-}" == "linux"* ]]; then
```

Comment:
using `$OSTYPE` instead of the target platform means that a macOS user can't cross-compile e2e_node.test for linux.